lib/runtime/unknown_transaction.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2022 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package runtime
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/ChainSafe/gossamer/pkg/scale"
+)
+
+// UnknownTransaction is child VDT of TransactionValidityError
+type UnknownTransaction scale.VaryingDataType
+
+// Index fulfils the VaryingDataTypeValue interface.  T
+func (UnknownTransaction) Index() uint {
+	return 1
+}
+
+var (
+	errLookupFailed      = errors.New("lookup failed")
+	errValidatorNotFound = errors.New("validator not found")
+	unknownCustom        UnknownCustom
+)
+
+// ValidityCannotLookup Could not look up some information that is required to validate the transaction
+type ValidityCannotLookup struct{}
+
+// Index Returns VDT index
+func (ValidityCannotLookup) Index() uint { return 0 }
+
+// NoUnsignedValidator No validator found for the given unsigned transaction
+type NoUnsignedValidator struct{}
+
+// Index Returns VDT index
+func (NoUnsignedValidator) Index() uint { return 1 }
+
+// UnknownCustom Any other custom unknown validity that is not covered
+type UnknownCustom uint8
+
+// Index Returns VDT index
+func (UnknownCustom) Index() uint { return 2 }
+
+func newUnknownError(data scale.VaryingDataTypeValue) error {
+	return fmt.Errorf("unknown error: %d", data)
```

Comment:
Is `%d` appropriate here? Shouldn't it be `%v` (or add a `String() string` method + `%s`)?